server/src/test/java/io/crate/execution/ddl/tables/CreateTableRequestTest.java

```diff
@@ -0,0 +1,118 @@
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.ddl.tables;
+
+import static io.crate.analyze.AnalyzedColumnDefinition.typeNameForESMapping;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.carrotsearch.hppc.IntArrayList;
+import io.crate.metadata.Reference;
+import io.crate.metadata.ReferenceIdent;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.SimpleReference;
+import io.crate.sql.tree.ColumnPolicy;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.types.DataTypes;
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+public class CreateTableRequestTest {
+
+    @Test
+    public void test_streaming() throws Exception {
+        RelationName rel = RelationName.of(QualifiedName.of("t1"), Schemas.DOC_SCHEMA_NAME);
+
+        Reference ref1 = new SimpleReference(
+            new ReferenceIdent(rel, "part_col_1"),
+            RowGranularity.DOC,
+            DataTypes.STRING,
+            1,
+            null,
+            null
+        );
+        Reference ref2 = new SimpleReference(
+            new ReferenceIdent(rel, "part_col_2"),
+            RowGranularity.DOC,
+            DataTypes.INTEGER,
+            2,
+            null,
+            null
+        );
+        Reference ref3 = new SimpleReference(
+            new ReferenceIdent(rel, "just_col"),
+            RowGranularity.DOC,
+            DataTypes.BYTE,
+            3,
+            null,
+            null
+        );
+        Reference ref4 = new SimpleReference(
+            new ReferenceIdent(rel, "some_routing_col"),
+            RowGranularity.DOC,
+            DataTypes.LONG,
+            4,
+            null,
+            null
+        );
+        List<Reference> refs = List.of(ref1, ref2, ref3, ref4);
+        List<String> partCol1 = List.of("part_col_1", typeNameForESMapping(DataTypes.STRING, "english", false));
```

Comment:
If the last parameter `isIndex` is false, the `2nd` parameter would not create any change in the type. See https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java#L490
```suggestion
        List<String> partCol1 = List.of("part_col_1", typeNameForESMapping(DataTypes.STRING, null, false));
```